components/brave_wallet/browser/asset_discovery_manager.cc

```diff
@@ -421,4 +703,39 @@ absl::optional<SolanaAddress> AssetDiscoveryManager::DecodeMintAddress(
   return SolanaAddress::FromBytes(pub_key_bytes);
 }
 
+// static
+// Creates a URL like
+// https://api.simplehash.com/api/v0/nfts/owners?chains={chains}&wallet_addresses={wallet_addresses}
+GURL AssetDiscoveryManager::GetSimpleHashNftsByWalletUrl(
+    const std::string& account_address,
+    const std::vector<std::string>& chain_ids) {
+  if (chain_ids.empty() || account_address.empty()) {
+    return GURL();
+  }
+
+  std::string urlStr =
+      base::StringPrintf("%s/api/v0/nfts/owners", kSimpleHashUrl);
+  const base::flat_map<std::string, std::string>& simple_hash_chain_ids =
+      ToSimpleHashChainId();
+  std::string chain_ids_param;
+  for (const auto& chain_id : chain_ids) {
+    auto it = simple_hash_chain_ids.find(chain_id);
+    if (it != simple_hash_chain_ids.end()) {
+      if (!chain_ids_param.empty()) {
+        chain_ids_param += ",";
+      }
+      chain_ids_param += it->second;
+    }
+  }
+
+  if (chain_ids_param.empty()) {
+    return GURL();
+  }
+
+  GURL url = GURL(urlStr);
+  url = net::AppendQueryParameter(url, "chains", chain_ids_param);
+  url = net::AppendQueryParameter(url, "wallet_addresses", account_address);
```

Comment:
Is there any concern about someone manipulating their wallet address here to append on additional query parameters here? I can't see any harm coming from them since it would just do some additional filtering to the API it looks like, but not sure if you took that into consideration.

If we wanted to prevent this we could just validate the wallet address before sticking it in this URL.

Reply:
I'm pretty sure by this point we're only working with valid addresses, but it wasn't obvious where that validation happens when I went to look.  I'll add some more explicitly for this class.

Reply:
Thanks, yeah when I was testing this originally it seemed like we did this on input specifically through the UI. The only ways that I could see that someone might be able to manipulate this would be at the filesystem layer which we've typically left out of scope. Only thought of it specifically for this case, because I saw there was a bunch of optional parameters we weren't using.

Reply:
Right, the interface frontends use is this one
```
interface BraveWalletService {
  DiscoverAssetsOnAllSupportedChains();
};
```
No account addresses are supplied.  Instead account addresses are [fetched from the keyring](https://github.com/brave/brave-core/blob/f1774b8b70cdc517824922ac6cacf4fbb04f7475/components/brave_wallet/browser/brave_wallet_service.cc#L1427-L1437), which are definitely valid.  I think fine to keep as is.

Reply:
Ok thanks for looking into this a bit more so resolving now.